src/util/panic_hook.rs

```diff
@@ -105,6 +112,10 @@ pub fn set_exit_hook(panic_abort: bool) {
         } else {
             orig_hook(info);
         }
+
+        // To collect remaining logs, drop guard before exit.
+        drop(LOG_GUARD.lock().unwrap().take());
```

Comment:
This doesn't seem like flushing the log. What if there is a thread that keeps writing logs?

Reply:
For the `Aync` Drainer, drop the guard will send a `AsyncMsg::Finish` which will stop the logger, it accepts no more logs, so `drop` returns eventually.

Reply:
Sounds hacky. Is there any more verbose and straightforward way?